### PR TITLE
Added Checks for proper file formats

### PR DESCRIPTION
**Pull request by tapananand**

### CODE REVIEW

File: MemeMaker.html [hunk 1]
```diff
@@ -74,6 +74,22 @@
             function saveFile() {
                 window.open( document.querySelector( 'canvas' ).toDataURL() );
             }
+            
+            function isValidFile(data)
+            {
+            	var mimeType = getMimeType(data);
+            	var allowedMimeTypes = ["image/jpeg", "image/jpg", "image/png", "image/gif"];
```
Comment (crisron-zz) [inline MemeMaker.html:80]: Can we add `bitmap` to the allowed file formats?

File: MemeMaker.html [hunk 2]
```diff
@@ -85,13 +101,19 @@
...
-                    console.log( fileObject.target.result );
+                    // Set image data to background image if it is a valid image (jpg/gif/png)
+                    if(isValidFile(data))
+                    	image.src = data;
+                    else	
+                    {
+                    	alert("Only jpg, png and gif images are allowed");
+                    	evt.target.value = "";
+                    }
+                    //console.log( fileObject.target.result );
```
Comment (crisron-zz) [inline MemeMaker.html]: Please remove this comment.

File: MemeMaker.html [hunk 3]
```diff
@@ -85,13 +101,19 @@
...
                         redrawMeme( window.imageSrc, null, null );
                     }
 
-                    // Set image data to background image
-                    image.src = data;
-                    console.log( fileObject.target.result );
+                    // Set image data to background image if it is a valid image (jpg/gif/png)
+                    if(isValidFile(data))
+                    	image.src = data;
+                    else
```
Comment (crisron-zz) [inline MemeMaker.html]: Please use K & R format for if...else block. Like this:

``` C
if( cond. ) {
    expr;
} else {
    expr1;
}
```

File: MemeMaker.html [hunk 4]
```diff
@@ -85,13 +100,18 @@
...
                     }
 
-                    // Set image data to background image
-                    image.src = data;
-                    console.log( fileObject.target.result );
+                    // Set image data to background image if it is a valid image (jpg/gif/png)
+                    if(isValidFile(data)) {
+                    	image.src = data;
+                    }
+                    else {
```
Comment (crisron-zz) [inline MemeMaker.html]: `else` block should start on the same line as the `if` closing brace.

File: MemeMaker.html (hunk 1 above)
Comment (tapananand) [inline MemeMaker.html:80]: Bitmap is not a standard format for the web, do you have a bmp image? May be can give a try by analyzing it's data uri

Comment (crisron-zz) [inline MemeMaker.html:80]: We can ignore that for now. Thanks for the patch. Going to merge it :+1:

**Pull request merged**